docs/bulk-retries-design.md

```diff
@@ -53,21 +65,26 @@ Because each message send is done in the context of a Satellite Receive operatio
 
 If there is a message in the `Forwarding` status when ServiceControl starts, then we don't know how many messages there are still in the staging queue to send. To counter this we turn on the satellite in Non-Counting mode. The idea for this is that the satellite will run until the queue is empty. Unfortunately there is nothing built into the Transport abstraction that allows us to query this so SC assumes that if it does not see any new messages from the `staging` queue within 45 seconds then it is empty.
 
+TODO: What happend when forwarding fails? Transport error, network error? What happens in the edge case then forwarding completes and we crash? Restart and finds an empty stating queue and times out?
```

Comment:
This is one benefit of the staging queue. It makes it more likely that if we can reach the broker to receive the message, then we can reach the broker to send it on.

> What happend when forwarding fails? Transport error, network error?

If forwarding fails for a message, then we count it, mark it as unresolved, and remove it from the batch. You should see warnings in the log that look like this:

```
Failed to send UNIQUE-MESSAGE-ID message to DESTINATION for retry. Attempting to revert message status to unresoved so it can be tried again.
```
The error information will be attached.

> What happens in the edge case then forwarding completes and we crash? Restart and finds an empty stating queue and times out?

Yes.